From e5a3a051f8cb32e5c8586b74f180c9a0b2d60e56 Mon Sep 17 00:00:00 2001
From: Dmitri Epshtein <dima@marvell.com>
Date: Tue, 16 Jan 2018 12:04:14 +0200
Subject: [PATCH 1/5] examples/ipsec-secgw: change config for 2 ports, 8 cpus

4 CPU cores per port
cfg	     SIP	     DIP	RXQ

ep0	192.168.106.10	192.168.171.10	 0
	192.168.106.10	192.168.176.10	 2
	192.168.106.10	192.168.180.10	 1
	192.168.106.10	192.168.175.10	 3

ep1	192.168.116.10	192.168.220.10	 2
	192.168.116.10	192.168.224.10	 0
	192.168.116.10	192.168.228.10	 3
	192.168.116.10	192.168.236.10	 1

Change-Id: Icb6989295a15453b370cc06ab6fbbf261075a6a7
Signed-off-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/49112
(cherry picked from commit 57250fc6bc8019ffd1625ac9c4c233c9811d8f0c)
Reviewed-on: http://vgitil04.il.marvell.com:8080/50203
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
---
 examples/ipsec-secgw/ep0.cfg | 182 ++++++++++++++++++++-----------------------
 examples/ipsec-secgw/ep1.cfg | 178 ++++++++++++++++++++----------------------
 2 files changed, 168 insertions(+), 192 deletions(-)

diff --git a/examples/ipsec-secgw/ep0.cfg b/examples/ipsec-secgw/ep0.cfg
index 299aa9e..a9e25aa 100644
--- a/examples/ipsec-secgw/ep0.cfg
+++ b/examples/ipsec-secgw/ep0.cfg
@@ -10,67 +10,57 @@
 #
 ###########################################################################
 
-#SP IPv4 rules
-sp ipv4 out esp protect 5 pri 1 dst 192.168.105.0/24 sport 0:65535 dport 0:65535
+# ************* SP IPv4 rules **************
+
+# IPv4 tunnel OUT modes
+sp ipv4 out esp protect 3 pri 1 dst 192.168.103.0/24 sport 0:65535 dport 0:65535
 sp ipv4 out esp protect 6 pri 1 dst 192.168.106.0/24 sport 0:65535 dport 0:65535
+
+# Transport OUT modes
 sp ipv4 out esp protect 10 pri 1 dst 192.168.175.0/24 sport 0:65535 dport 0:65535
 sp ipv4 out esp protect 11 pri 1 dst 192.168.176.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 12 pri 1 dst 192.168.171.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 13 pri 1 dst 192.168.180.0/24 sport 0:65535 dport 0:65535
+
+# Null IPv4 tunnel OUT mode
 sp ipv4 out esp protect 15 pri 1 dst 192.168.200.0/24 sport 0:65535 dport 0:65535
 sp ipv4 out esp protect 16 pri 1 dst 192.168.201.0/24 sport 0:65535 dport 0:65535
-sp ipv4 out esp protect 25 pri 1 dst 192.168.55.0/24 sport 0:65535 dport 0:65535
-sp ipv4 out esp protect 26 pri 1 dst 192.168.56.0/24 sport 0:65535 dport 0:65535
+
+# Bypass OUT mode
 sp ipv4 out esp bypass pri 1 dst 192.168.240.0/24 sport 0:65535 dport 0:65535
 sp ipv4 out esp bypass pri 1 dst 192.168.241.0/24 sport 0:65535 dport 0:65535
 
-sp ipv4 in esp protect 105 pri 1 dst 192.168.115.0/24 sport 0:65535 dport 0:65535
+# IPv4 tunnel IN modes
+sp ipv4 in esp protect 103 pri 1 dst 192.168.113.0/24 sport 0:65535 dport 0:65535
 sp ipv4 in esp protect 106 pri 1 dst 192.168.116.0/24 sport 0:65535 dport 0:65535
-sp ipv4 in esp protect 110 pri 1 dst 192.168.185.0/24 sport 0:65535 dport 0:65535
-sp ipv4 in esp protect 111 pri 1 dst 192.168.186.0/24 sport 0:65535 dport 0:65535
+
+# Transport IN modes
+sp ipv4 in esp protect 110 pri 1 dst 192.168.220.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 111 pri 1 dst 192.168.224.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 112 pri 1 dst 192.168.228.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 113 pri 1 dst 192.168.236.0/24 sport 0:65535 dport 0:65535
+
+# Null IPv4 tunnel IN mode
 sp ipv4 in esp protect 115 pri 1 dst 192.168.210.0/24 sport 0:65535 dport 0:65535
 sp ipv4 in esp protect 116 pri 1 dst 192.168.211.0/24 sport 0:65535 dport 0:65535
-sp ipv4 in esp protect 115 pri 1 dst 192.168.210.0/24 sport 0:65535 dport 0:65535
-sp ipv4 in esp protect 125 pri 1 dst 192.168.65.0/24 sport 0:65535 dport 0:65535
-sp ipv4 in esp protect 125 pri 1 dst 192.168.65.0/24 sport 0:65535 dport 0:65535
-sp ipv4 in esp protect 126 pri 1 dst 192.168.66.0/24 sport 0:65535 dport 0:65535
+
+# Bypass IN mode
 sp ipv4 in esp bypass pri 1 dst 192.168.245.0/24 sport 0:65535 dport 0:65535
 sp ipv4 in esp bypass pri 1 dst 192.168.246.0/24 sport 0:65535 dport 0:65535
 
-#SP IPv6 rules
-sp ipv6 out esp protect 5 pri 1 dst 0000:0000:0000:0000:5555:5555:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 6 pri 1 dst 0000:0000:0000:0000:6666:6666:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 10 pri 1 dst 0000:0000:1111:1111:0000:0000:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 11 pri 1 dst 0000:0000:1111:1111:1111:1111:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 25 pri 1 dst 0000:0000:0000:0000:aaaa:aaaa:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 26 pri 1 dst 0000:0000:0000:0000:bbbb:bbbb:0000:0000/96 \
-sport 0:65535 dport 0:65535
-
-sp ipv6 in esp protect 15 pri 1 dst ffff:0000:0000:0000:5555:5555:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 16 pri 1 dst ffff:0000:0000:0000:6666:6666:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 110 pri 1 dst ffff:0000:1111:1111:0000:0000:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 111 pri 1 dst ffff:0000:1111:1111:1111:1111:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 125 pri 1 dst ffff:0000:0000:0000:aaaa:aaaa:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 126 pri 1 dst ffff:0000:0000:0000:bbbb:bbbb:0000:0000/96 \
-sport 0:65535 dport 0:65535
-
-#SA rules
-sa out 5 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+
+# ******** SA rules ***********
+
+# IPv4 tunnel OUT modes
+sa out 3 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
 auth_algo sha1-hmac auth_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
-mode ipv4-tunnel src 172.16.1.5 dst 172.16.2.5
+mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.5
 
 sa out 6 cipher_algo aes-128-cbc cipher_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
 a0:a0:a0:a0:a0 auth_algo sha1-hmac auth_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
 a0:a0:a0:a0:a0:a0:a0:a0:a0 mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.6
 
+# Transport OUT modes
 sa out 10 cipher_algo aes-128-cbc cipher_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
 a1:a1:a1:a1:a1 auth_algo sha1-hmac auth_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
 a1:a1:a1:a1:a1:a1:a1:a1:a1 mode transport
@@ -79,32 +69,28 @@ sa out 11 cipher_algo aes-128-cbc cipher_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
 b2:b2:b2:b2:b2 auth_algo sha1-hmac auth_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
 b2:b2:b2:b2:b2:b2:b2:b2:b2 mode transport
 
-sa out 15 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.5 \
-dst 172.16.2.5
-
-sa out 16 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.6 \
-dst 172.16.2.6
-
-sa out 25 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+sa out 12 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
 c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
-c3:c3:c3:c3:c3:c3:c3:c3:c3 mode ipv6-tunnel \
-src 1111:1111:1111:1111:1111:1111:1111:5555 \
-dst 2222:2222:2222:2222:2222:2222:2222:5555
+c3:c3:c3:c3:c3:c3:c3:c3:c3 mode transport
+
+sa out 13 cipher_algo aes-128-cbc cipher_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4 auth_algo sha1-hmac auth_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4:d4:d4:d4:d4 mode transport
 
-sa out 26 cipher_algo aes-128-cbc cipher_key 4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:\
-4d:4d:4d:4d:4d auth_algo sha1-hmac auth_key 4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:\
-4d:4d:4d:4d:4d:4d:4d:4d:4d mode ipv6-tunnel \
-src 1111:1111:1111:1111:1111:1111:1111:6666 \
-dst 2222:2222:2222:2222:2222:2222:2222:6666
+# Null IPv4 tunnel OUT mode
+sa out 15 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.5 dst 172.16.2.5
+sa out 16 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.6
 
-sa in 105 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+# IPv4 tunnel IN modes
+sa in 103 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
 auth_algo sha1-hmac auth_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
-mode ipv4-tunnel src 172.16.2.5 dst 172.16.1.5
+mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.5
 
 sa in 106 cipher_algo aes-128-cbc cipher_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
 a0:a0:a0:a0:a0 auth_algo sha1-hmac auth_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
 a0:a0:a0:a0:a0:a0:a0:a0:a0 mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.6
 
+# Transport IN modes
 sa in 110 cipher_algo aes-128-cbc cipher_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
 a1:a1:a1:a1:a1 auth_algo sha1-hmac auth_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
 a1:a1:a1:a1:a1:a1:a1:a1:a1 mode transport
@@ -113,48 +99,50 @@ sa in 111 cipher_algo aes-128-cbc cipher_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
 b2:b2:b2:b2:b2 auth_algo sha1-hmac auth_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
 b2:b2:b2:b2:b2:b2:b2:b2:b2 mode transport
 
-sa in 115 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.5 \
-dst 172.16.1.5
+sa in 112 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3:c3:c3:c3:c3 mode transport
+
+sa in 113 cipher_algo aes-128-cbc cipher_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4 auth_algo sha1-hmac auth_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4:d4:d4:d4:d4 mode transport
 
+# Null IPv4 tunnel IN mode
+sa in 115 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.5 dst 172.16.1.5
 sa in 116 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.6
 
-sa in 125 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
-c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
-c3:c3:c3:c3:c3:c3:c3:c3:c3 mode ipv6-tunnel \
-src 2222:2222:2222:2222:2222:2222:2222:5555 \
-dst 1111:1111:1111:1111:1111:1111:1111:5555
-
-sa in 126 cipher_algo aes-128-cbc cipher_key 4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:\
-4d:4d:4d:4d:4d auth_algo sha1-hmac auth_key 4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:\
-4d:4d:4d:4d:4d:4d:4d:4d:4d mode ipv6-tunnel \
-src 2222:2222:2222:2222:2222:2222:2222:6666 \
-dst 1111:1111:1111:1111:1111:1111:1111:6666
-
-#Routing rules
-rt ipv4 dst 172.16.2.5/32 port 0
+
+# ********** Routing rules **************
+
+# IPv4 tunnel OUT rules
+rt ipv4 dst 172.16.2.5/32 port 1
 rt ipv4 dst 172.16.2.6/32 port 1
-rt ipv4 dst 192.168.175.0/24 port 0
+
+# Transport OUT rules
+rt ipv4 dst 192.168.175.0/24 port 1
 rt ipv4 dst 192.168.176.0/24 port 1
-rt ipv4 dst 192.168.240.0/24 port 0
+rt ipv4 dst 192.168.171.0/24 port 1
+rt ipv4 dst 192.168.180.0/24 port 1
+
+# Bypass OUT mode
+rt ipv4 dst 192.168.240.0/24 port 1
 rt ipv4 dst 192.168.241.0/24 port 1
-rt ipv4 dst 192.168.115.0/24 port 2
-rt ipv4 dst 192.168.116.0/24 port 3
-rt ipv4 dst 192.168.65.0/24 port 2
-rt ipv4 dst 192.168.66.0/24 port 3
-rt ipv4 dst 192.168.185.0/24 port 2
-rt ipv4 dst 192.168.186.0/24 port 3
-rt ipv4 dst 192.168.210.0/24 port 2
-rt ipv4 dst 192.168.211.0/24 port 3
-rt ipv4 dst 192.168.245.0/24 port 2
-rt ipv4 dst 192.168.246.0/24 port 3
-
-rt ipv6 dst 2222:2222:2222:2222:2222:2222:2222:5555/116 port 0
-rt ipv6 dst 2222:2222:2222:2222:2222:2222:2222:6666/116 port 1
-rt ipv6 dst 0000:0000:1111:1111:0000:0000:0000:0000/116 port 0
-rt ipv6 dst 0000:0000:1111:1111:1111:1111:0000:0000/116 port 1
-rt ipv6 dst ffff:0000:0000:0000:aaaa:aaaa:0000:0000/116 port 2
-rt ipv6 dst ffff:0000:0000:0000:bbbb:bbbb:0000:0000/116 port 3
-rt ipv6 dst ffff:0000:0000:0000:5555:5555:0000:0000/116 port 2
-rt ipv6 dst ffff:0000:0000:0000:6666:6666:0000:0000/116 port 3
-rt ipv6 dst ffff:0000:1111:1111:0000:0000:0000:0000/116 port 2
-rt ipv6 dst ffff:0000:1111:1111:1111:1111:0000:0000/116 port 3
+
+# IPv4 tunnel IN rules
+rt ipv4 dst 192.168.113.0/24 port 0
+rt ipv4 dst 192.168.116.0/24 port 0
+
+# Transport IN rules
+rt ipv4 dst 192.168.220.0/24 port 0
+rt ipv4 dst 192.168.224.0/24 port 0
+rt ipv4 dst 192.168.228.0/24 port 0
+rt ipv4 dst 192.168.236.0/24 port 0
+
+# Null IPv4 tunnel IN mode
+rt ipv4 dst 192.168.210.0/24 port 0
+rt ipv4 dst 192.168.211.0/24 port 0
+
+# Bypass IN mode
+rt ipv4 dst 192.168.245.0/24 port 0
+rt ipv4 dst 192.168.246.0/24 port 0
+
diff --git a/examples/ipsec-secgw/ep1.cfg b/examples/ipsec-secgw/ep1.cfg
index 3f6ff81..325078f 100644
--- a/examples/ipsec-secgw/ep1.cfg
+++ b/examples/ipsec-secgw/ep1.cfg
@@ -10,60 +10,49 @@
 #
 ###########################################################################
 
-#SP IPv4 rules
-sp ipv4 in esp protect 5 pri 1 dst 192.168.105.0/24 sport 0:65535 dport 0:65535
+# *********** SP IPv4 rules ***************
+
+# IPv4 tunnel IN modes
+sp ipv4 in esp protect 3 pri 1 dst 192.168.103.0/24 sport 0:65535 dport 0:65535
 sp ipv4 in esp protect 6 pri 1 dst 192.168.106.0/24 sport 0:65535 dport 0:65535
+
+# Transport IN modes
 sp ipv4 in esp protect 10 pri 1 dst 192.168.175.0/24 sport 0:65535 dport 0:65535
 sp ipv4 in esp protect 11 pri 1 dst 192.168.176.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 12 pri 1 dst 192.168.171.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 13 pri 1 dst 192.168.180.0/24 sport 0:65535 dport 0:65535
+
+# Null IPv4 tunnel IN mode
 sp ipv4 in esp protect 15 pri 1 dst 192.168.200.0/24 sport 0:65535 dport 0:65535
 sp ipv4 in esp protect 16 pri 1 dst 192.168.201.0/24 sport 0:65535 dport 0:65535
-sp ipv4 in esp protect 25 pri 1 dst 192.168.55.0/24 sport 0:65535 dport 0:65535
-sp ipv4 in esp protect 26 pri 1 dst 192.168.56.0/24 sport 0:65535 dport 0:65535
+
+# Bypass IN mode
 sp ipv4 in esp bypass dst 192.168.240.0/24 sport 0:65535 dport 0:65535
 sp ipv4 in esp bypass dst 192.168.241.0/24 sport 0:65535 dport 0:65535
 
-sp ipv4 out esp protect 105 pri 1 dst 192.168.115.0/24 sport 0:65535 dport 0:65535
+# IPv4 tunnel OUT modes
+sp ipv4 out esp protect 103 pri 1 dst 192.168.113.0/24 sport 0:65535 dport 0:65535
 sp ipv4 out esp protect 106 pri 1 dst 192.168.116.0/24 sport 0:65535 dport 0:65535
-sp ipv4 out esp protect 110 pri 1 dst 192.168.185.0/24 sport 0:65535 dport 0:65535
-sp ipv4 out esp protect 111 pri 1 dst 192.168.186.0/24 sport 0:65535 dport 0:65535
+
+# Transport OUT modes
+sp ipv4 out esp protect 110 pri 1 dst 192.168.220.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 111 pri 1 dst 192.168.224.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 112 pri 1 dst 192.168.228.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 113 pri 1 dst 192.168.236.0/24 sport 0:65535 dport 0:65535
+
+# Null IPv4 tunnel OUT mode
 sp ipv4 out esp protect 115 pri 1 dst 192.168.210.0/24 sport 0:65535 dport 0:65535
 sp ipv4 out esp protect 116 pri 1 dst 192.168.211.0/24 sport 0:65535 dport 0:65535
-sp ipv4 out esp protect 115 pri 1 dst 192.168.210.0/24 sport 0:65535 dport 0:65535
-sp ipv4 out esp protect 125 pri 1 dst 192.168.65.0/24 sport 0:65535 dport 0:65535
-sp ipv4 out esp protect 125 pri 1 dst 192.168.65.0/24 sport 0:65535 dport 0:65535
-sp ipv4 out esp protect 126 pri 1 dst 192.168.66.0/24 sport 0:65535 dport 0:65535
+
+# Bypass OUT mode
 sp ipv4 out esp bypass pri 1 dst 192.168.245.0/24 sport 0:65535 dport 0:65535
 sp ipv4 out esp bypass pri 1 dst 192.168.246.0/24 sport 0:65535 dport 0:65535
 
-#SP IPv6 rules
-sp ipv6 in esp protect 5 pri 1 dst 0000:0000:0000:0000:5555:5555:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 6 pri 1 dst 0000:0000:0000:0000:6666:6666:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 10 pri 1 dst 0000:0000:1111:1111:0000:0000:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 11 pri 1 dst 0000:0000:1111:1111:1111:1111:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 25 pri 1 dst 0000:0000:0000:0000:aaaa:aaaa:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 in esp protect 26 pri 1 dst 0000:0000:0000:0000:bbbb:bbbb:0000:0000/96 \
-sport 0:65535 dport 0:65535
-
-sp ipv6 out esp protect 15 pri 1 dst ffff:0000:0000:0000:5555:5555:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 16 pri 1 dst ffff:0000:0000:0000:6666:6666:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 110 pri 1 dst ffff:0000:1111:1111:0000:0000:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 111 pri 1 dst ffff:0000:1111:1111:1111:1111:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 125 pri 1 dst ffff:0000:0000:0000:aaaa:aaaa:0000:0000/96 \
-sport 0:65535 dport 0:65535
-sp ipv6 out esp protect 126 pri 1 dst ffff:0000:0000:0000:bbbb:bbbb:0000:0000/96 \
-sport 0:65535 dport 0:65535
-
-#SA rules
-sa in 5 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+
+# ************* SA rules ****************
+
+# IPv4 tunnel IN modes
+sa in 3 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
 auth_algo sha1-hmac auth_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
 mode ipv4-tunnel src 172.16.1.5 dst 172.16.2.5
 
@@ -71,6 +60,7 @@ sa in 6 cipher_algo aes-128-cbc cipher_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
 a0:a0:a0:a0:a0 auth_algo sha1-hmac auth_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
 a0:a0:a0:a0:a0:a0:a0:a0:a0 mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.6
 
+# Transport IN modes
 sa in 10 cipher_algo aes-128-cbc cipher_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
 a1:a1:a1:a1:a1 auth_algo sha1-hmac auth_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
 a1:a1:a1:a1:a1:a1:a1:a1:a1 mode transport
@@ -79,24 +69,19 @@ sa in 11 cipher_algo aes-128-cbc cipher_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
 b2:b2:b2:b2:b2 auth_algo sha1-hmac auth_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
 b2:b2:b2:b2:b2:b2:b2:b2:b2 mode transport
 
-sa in 15 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.5 \
-dst 172.16.2.5
-
-sa in 16 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.6 \
-dst 172.16.2.6
-
-sa in 25 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+sa in 12 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
 c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
-c3:c3:c3:c3:c3:c3:c3:c3:c3 mode ipv6-tunnel \
-src 1111:1111:1111:1111:1111:1111:1111:5555 \
-dst 2222:2222:2222:2222:2222:2222:2222:5555
+c3:c3:c3:c3:c3:c3:c3:c3:c3 mode transport
+
+sa in 13 cipher_algo aes-128-cbc cipher_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4 auth_algo sha1-hmac auth_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4:d4:d4:d4:d4 mode transport
 
-sa in 26 cipher_algo aes-128-cbc cipher_key 4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:\
-4d:4d:4d:4d:4d auth_algo sha1-hmac auth_key 4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:\
-4d:4d:4d:4d:4d:4d:4d:4d:4d mode ipv6-tunnel \
-src 1111:1111:1111:1111:1111:1111:1111:6666 \
-dst 2222:2222:2222:2222:2222:2222:2222:6666
+# Null IPv4 tunnel IN mode
+sa in 15 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.5 dst 172.16.2.5
+sa in 16 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.6
 
+# IPv4 tunnel OUT modes
 sa out 105 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
 auth_algo sha1-hmac auth_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
 mode ipv4-tunnel src 172.16.2.5 dst 172.16.1.5
@@ -105,6 +90,7 @@ sa out 106 cipher_algo aes-128-cbc cipher_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
 a0:a0:a0:a0:a0 auth_algo sha1-hmac auth_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
 a0:a0:a0:a0:a0:a0:a0:a0:a0 mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.6
 
+# Transport OUT modes
 sa out 110 cipher_algo aes-128-cbc cipher_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
 a1:a1:a1:a1:a1 auth_algo sha1-hmac auth_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
 a1:a1:a1:a1:a1:a1:a1:a1:a1 mode transport
@@ -113,48 +99,50 @@ sa out 111 cipher_algo aes-128-cbc cipher_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
 b2:b2:b2:b2:b2 auth_algo sha1-hmac auth_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
 b2:b2:b2:b2:b2:b2:b2:b2:b2 mode transport
 
-sa out 115 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.5 \
-dst 172.16.1.5
+sa out 112 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3:c3:c3:c3:c3 mode transport
+
+sa out 113 cipher_algo aes-128-cbc cipher_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4 auth_algo sha1-hmac auth_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4:d4:d4:d4:d4 mode transport
 
+# Null IPv4 tunnel OUT mode
+sa out 115 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.5 dst 172.16.1.5
 sa out 116 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.6
 
-sa out 125 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
-c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
-c3:c3:c3:c3:c3:c3:c3:c3:c3 mode ipv6-tunnel \
-src 2222:2222:2222:2222:2222:2222:2222:5555 \
-dst 1111:1111:1111:1111:1111:1111:1111:5555
-
-sa out 126 cipher_algo aes-128-cbc cipher_key 4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:\
-4d:4d:4d:4d:4d auth_algo sha1-hmac auth_key 4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:4d:\
-4d:4d:4d:4d:4d:4d:4d:4d:4d mode ipv6-tunnel \
-src 2222:2222:2222:2222:2222:2222:2222:6666 \
-dst 1111:1111:1111:1111:1111:1111:1111:6666
-
-#Routing rules
-rt ipv4 dst 172.16.1.5/32 port 0
+
+# ********* Routing rules **********
+
+# IPv4 tunnel OUT rules
+rt ipv4 dst 172.16.1.5/32 port 1
 rt ipv4 dst 172.16.1.6/32 port 1
-rt ipv4 dst 192.168.185.0/24 port 0
-rt ipv4 dst 192.168.186.0/24 port 1
-rt ipv4 dst 192.168.245.0/24 port 0
+
+# Transport OUT rules
+rt ipv4 dst 192.168.220.0/24 port 1
+rt ipv4 dst 192.168.224.0/24 port 1
+rt ipv4 dst 192.168.228.0/24 port 1
+rt ipv4 dst 192.168.236.0/24 port 1
+
+# Bypass OUT mode
+rt ipv4 dst 192.168.245.0/24 port 1
 rt ipv4 dst 192.168.246.0/24 port 1
-rt ipv4 dst 192.168.105.0/24 port 2
-rt ipv4 dst 192.168.106.0/24 port 3
-rt ipv4 dst 192.168.55.0/24 port 2
-rt ipv4 dst 192.168.56.0/24 port 3
-rt ipv4 dst 192.168.175.0/24 port 2
-rt ipv4 dst 192.168.176.0/24 port 3
-rt ipv4 dst 192.168.200.0/24 port 2
-rt ipv4 dst 192.168.201.0/24 port 3
-rt ipv4 dst 192.168.240.0/24 port 2
-rt ipv4 dst 192.168.241.0/24 port 3
-
-rt ipv6 dst 1111:1111:1111:1111:1111:1111:1111:5555/116 port 0
-rt ipv6 dst 1111:1111:1111:1111:1111:1111:1111:6666/116 port 1
-rt ipv6 dst ffff:0000:1111:1111:0000:0000:0000:0000/116 port 0
-rt ipv6 dst ffff:0000:1111:1111:1111:1111:0000:0000/116 port 1
-rt ipv6 dst 0000:0000:0000:0000:aaaa:aaaa:0000:0000/116 port 2
-rt ipv6 dst 0000:0000:0000:0000:bbbb:bbbb:0000:0000/116 port 3
-rt ipv6 dst 0000:0000:0000:0000:5555:5555:0000:0000/116 port 2
-rt ipv6 dst 0000:0000:0000:0000:6666:6666:0000:0000/116 port 3
-rt ipv6 dst 0000:0000:1111:1111:0000:0000:0000:0000/116 port 2
-rt ipv6 dst 0000:0000:1111:1111:1111:1111:0000:0000/116 port 3
+
+# IPv4 tunnel IN rules
+rt ipv4 dst 192.168.103.0/24 port 0
+rt ipv4 dst 192.168.106.0/24 port 0
+
+# Transport IN rules
+rt ipv4 dst 192.168.175.0/24 port 0
+rt ipv4 dst 192.168.176.0/24 port 0
+rt ipv4 dst 192.168.171.0/24 port 0
+rt ipv4 dst 192.168.180.0/24 port 0
+
+# Null IPv4 tunnel IN mode
+rt ipv4 dst 192.168.200.0/24 port 0
+rt ipv4 dst 192.168.201.0/24 port 0
+
+# Bypass IN mode
+rt ipv4 dst 192.168.240.0/24 port 0
+rt ipv4 dst 192.168.241.0/24 port 0
+
-- 
2.7.4

